src_c/surface.c

```diff
@@ -1907,23 +1907,37 @@ surf_blits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
     PyObject *special_flags = NULL;
     PyObject *ret = NULL;
     PyObject *retrect = NULL;
-    Py_ssize_t itemlength;
+    Py_ssize_t itemlength, sequencelength, curriter = 0;
     int doreturn = 1;
     int bliterrornum = 0;
+    int issequence = 0;
     static char *kwids[] = {"blit_sequence", "doreturn", NULL};
     if (!PyArg_ParseTupleAndKeywords(args, keywds, "O|i", kwids, &blitsequence,
                                      &doreturn))
         return NULL;
 
+    if (!PyIter_Check(blitsequence) &&
+        !(issequence = PySequence_Check(blitsequence))) {
+        bliterrornum = BLITS_ERR_SEQUENCE_REQUIRED;
+        goto bliterror;
+    }
+
     if (doreturn) {
-        ret = PyList_New(0);
+        if (issequence) {
+            sequencelength = PySequence_Size(blitsequence);
+            if (sequencelength == -1) {
+                bliterrornum = BLITS_ERR_SEQUENCE_REQUIRED;
```

Comment:
```suggestion
                bliterrornum = BLITS_ERR_PY_EXCEPTION_RAISED;
```
Because `PySequence_Size` would have already set something

Alternatively, there can be a simple `return NULL;` too here to exit early (because no other variables that need cleanup are initialised at this point)

Reply:
I'll go with the return NULL way